src/borrower/feed/navfeedPV.sol

```diff
@@ -0,0 +1,222 @@
+        return uint128(value);
+    }
+    
+    // returns the ceiling of a loan
+    // the ceiling defines the maximum amount which can be borrowed
+    function ceiling(uint loan) public virtual view returns (uint) {
+        bytes32 nftID_ = nftID(loan);
+        uint initialCeiling = rmul(nftValues(nftID_), ceilingRatio(risk(nftID_)));
+        return safeSub(initialCeiling, pile.debt(loan));
+    }
+
+    // --- Administration ---
+    function depend(bytes32 contractName, address addr) external auth {
+        if (contractName == "pile") {pile = PileLike(addr);}
+        else if (contractName == "shelf") { shelf = ShelfLike(addr); }
+        else revert();
+        emit Depend(contractName, addr);
+    }
+
+    function file(bytes32 name, uint risk_, uint thresholdRatio_, uint ceilingRatio_, uint rate_) public auth {
+        if(name == "riskGroup") {
+            require(ceilingRatio(risk_) == 0, "risk-group-in-usage");
+            riskGroup[risk_].thresholdRatio = toUint128(thresholdRatio_);
+            riskGroup[risk_].ceilingRatio = toUint128(ceilingRatio_);
+
+            // set interestRate for risk group
+            pile.file("rate", risk_, rate_);
+            emit File(name, risk_, thresholdRatio_, ceilingRatio_, rate_);
+
+        } else { revert ("unknown name");}
+    }
+
+    // --- Actions ---
+    function borrow(uint loan, uint amount) external virtual auth returns(uint navIncrease) {
+        require(ceiling(loan) >= amount, "borrow-amount-too-high");
+        return amount;
+    }
+
+    function repay(uint loan, uint amount) external virtual auth {
+    }
+
+    function borrowEvent(uint loan, uint) public virtual auth {
+        uint risk_ = risk(nftID(loan));
+        // when issued every loan has per default interest rate of risk group 0.
+        // correct interest rate has to be set on first borrow event
+        if (pile.loanRates(loan) != risk_) {
+            // set loan interest rate to the one of the correct risk group
+            pile.setRate(loan, risk_);
+        }
+    }
+
+    function repayEvent(uint loan, uint amount) public virtual auth {}
+    function lockEvent(uint loan) public virtual auth {}
+    function unlockEvent(uint loan) public virtual auth {}
+
+    function writeOff(uint loan) public auth {
+       pile.changeRate(loan, WRITEOFF_RATE_GROUP);
+    }
+
+    function isLoanWrittenOff(uint loan) public view returns(bool) {
+        return pile.loanRates(loan) == WRITEOFF_RATE_GROUP;
+    }
+
+    // --- NAV calculation ---
+    function currentNAV() public view returns(uint) {
+        uint totalDebt;
+        // calculate total debt
+        for (uint loanId = 1; loanId <= shelf.loanCount(); loanId++) {
+            totalDebt = safeAdd(totalDebt, pile.debt(loanId));
+        }
+
+        // substract writtenoff loans -> all writtenOff loans are moved to writeOffRateGroup
+        totalDebt = safeSub(totalDebt, pile.rateDebt(WRITEOFF_RATE_GROUP));
```

Comment:
If you iterate over all loan debt it make sense to remove it from the total debt.

However, the write-offs should still be added to the currentNAV in my opinion.
```
Example:
two loans:
1. debt: 100 DAI, riskGroup=1
2. debt: 80 DAI, writeOffGroup; writeOffFactor= 0.5 
```

For me, it makes sense to remove 80 DAI but you still need to add 80 DAI * 0.5 = 40 DAI to the NAV?

It would only make sense if the writeOff is always 100%. Is this the case?